aptos-move/aptos-vm/src/delta_ext/transaction.rs

```diff
@@ -65,6 +67,46 @@ impl TransactionOutputExt {
     pub fn into(self) -> (DeltaChangeSet, TransactionOutput) {
         (self.delta_change_set, self.output)
     }
+
+    /// Similar to `into()` but tries to apply delta changes as well.
+    pub fn into_transaction_output_with_status(
+        self,
+        state_view: &impl StateView,
+    ) -> (VMStatus, TransactionOutput) {
+        let (delta_change_set, txn_output) = self.into();
+
+        // No deltas - return immediately.
+        if delta_change_set.is_empty() {
+            return (VMStatus::Executed, txn_output);
+        }
+
+        match delta_change_set.try_into_write_set_mut(state_view) {
+            Err(_) => {
+                // TODO: at this point we know that delta application failed
+                // (and it should have occurred in user transaction in general).
+                // We need to rerun the epilogue and charge gas. Since we
+                // support only a limited set of features for the aggregator
+                // anyway, for now - panic.
+                panic!("something terrible happened when applying aggregator deltas")
```

Comment:
@gelash @zekun since we will use aggregator for total gas supply for V1, all deltas will be only created in the epilogue. So here error should not be really possible. Did you have something similar in mind (in terms of design)?

Alternative: we return Result instead and on caller side (which has access to VM) panic or rerun the epilogue